src/gsd.js

```diff
@@ -1,177 +1,219 @@
-var Opt = require("ml-optimize-lorentzian");
-var stats = require("ml-stat");
-var extend = require('extend');
-var SG = require('ml-savitzky-golay-generalized');
-
-var sgDefOptions = {
-    windowSize: 9,
-    polynomial: 3
-};
+'use strict';
 
+const extend = require('extend');
+const SG = require('ml-savitzky-golay-generalized');
 
-function gsd(x, yIn, options){
-    //options = extend({}, defaultOptions, options);
-    var options=Object.create(options || {});
-    if (options.minMaxRatio===undefined) options.minMaxRatio=0.00025;
-    if (options.broadRatio===undefined) options.broadRatio=0.00;
-    if (options.noiseLevel===undefined) options.noiseLevel=undefined;
-    if (options.noiseFactor===undefined) options.noiseFactor=3;
```

Comment:
Yes, you may remove that parameter. The factor is there because before the noise level was calculated inside, and the only way to control it was through the noiseFactor.